### PR TITLE
Add cloud script support

### PR DESCRIPTION
**Pull request by mofirouz**

### CODE REVIEW

**Comment by mofirouz** [comment]
Need to set ScriptID in the test cases

File: Classes/HLClient.h [hunk 1]
```diff
@@ -68,6 +68,8 @@
  */
 +(PMKPromise*)getLeaderboards;
 
++(PMKPromise*)executeScript:(NSString*)scriptId withPayload:(id)json;
```
Comment (novabyte) [inline Classes/HLClient.h]: Missing SDK reference documentation.

File: Tests/HLClientTest.m [hunk 2]
```diff
@@ -42,6 +42,7 @@ @implementation HLClientTest
 id achievementId = @"ec6764eadd274b9298887de9f5da0a5e";
 
 + (void)setUp {
+    [Expecta setAsynchronousTestTimeout:30];
```
Comment (novabyte) [inline Tests/HLClientTest.m]: I suggest we adjust this value to `10` per test case.

File: Tests/HLSessionClientTest.m [hunk 3]
```diff
@@ -52,8 +52,10 @@ @implementation HLSessionClientTest
 id gameLeaderboardId = @"5141dd1c31354741967e77f409ce755e";
 id matchTurnData = @"MatchTurnData";
 id productId = @"some.purchased.product.id";
+id scriptId = @"28b7cb10af864361b48bc437ff2fc6b9";
 
 + (void)setUp {
+    [Expecta setAsynchronousTestTimeout:30];
```
Comment (novabyte) [inline Tests/HLSessionClientTest.m]: Same as above.

**Comment by novabyte** [comment]
Missing a test case for `HLClient` with script execution. I think it's also worth adding the `iosunittest.js` script to the Tests folder so we have it as a reference in the future.

**Comment by novabyte** [comment]
@mofirouz Looks awesome! :+1:

**Pull request closed without merging**